src/logger/HttpLogger.go

```diff
@@ -0,0 +1,67 @@
+package logger
+
+import (
+	"strings"
+)
+
+const httpLoggerAgent string = "HttpLogger.go"
+
+//base HttpLogger definition
+type HttpLogger struct {
+	*BaseLogger
+	rules HttpRules
+}
+
+// initialize HttpLogger
+func NewHttpLogger(options Options) *HttpLogger {
+	baseLogger := NewBaseLogger(options.agent, options.url, options.enabled, options.queue)
+
+	loggerRules := NewHttpRules(options.rules)
+
+	logger := &HttpLogger{
+		baseLogger,
+		loggerRules,
+	}
+
+	logger.skipCompression = loggerRules.skipCompression
+	logger.skipSubmission = loggerRules.skipSubmission
+
+	if (logger.url != "") && (strings.HasPrefix(logger.url, "http:") && !logger.rules.allowHttpUrl) {
+		logger.enableable = false
+		logger.enabled = false
+	}
+
+	return logger
+}
+
+// getter for rules
+func (logger *HttpLogger) Rules() *HttpRules {
+	return logger.rules
+}
+
+func (logger *HttpLogger) submitIfPassing(details [][]string) {
+	details = logger.rules.apply(details)
+
+	if details == nil {
+		return
+	}
+
+	details = append(details, []string{"host", logger.host})
+
+	logger.submit(msgStringify(details))
+}
+
+// method for converting message details to string format
+func msgStringify(msg [][]string) string {
+	stringified := ""
+	n := len(msg)
+	for i, val := range msg {
+		stringified += "[" + strings.Join(val, ", ") + "]"
+		if i != n-1 {
+			stringified += ","
+		}
+	}
+	stringified = "[" + stringified + "]"
+
+	return stringified
+}
```

Comment:
This was pretty quick since I helped build it.  I think the options helps us pass everything we need in one go.